### PR TITLE
gtkui: (re)fix trkproperties to allow the deletion of fields longer than 4,999 characters

### PR DESCRIPTION
**Pull request by Lithopsian**
I think this was done once, but it isn't working now.

### CODE REVIEW

**Comment by Alexey-Yakovenko** [comment]
I don't remember merging this before.. what makes you think it happened?

File: plugins/gtkui/trkproperties.c [hunk 1]
```diff
@@ -266,7 +266,7 @@ on_metadata_edited (GtkCellRendererText *renderer, gchar *path, gchar *new_text,
     gtk_tree_model_get_value (GTK_TREE_MODEL (store), &iter, 3, &mult);
     const char *svalue = g_value_get_string (&value);
     int imult = g_value_get_int (&mult);
-    if (strcmp (svalue, new_text) && (!imult || strlen (new_text))) {
+    if (strcmp (svalue, new_text) && (!imult || strlen (new_text) == 0)) {
```
Comment (Alexey-Yakovenko) [inline plugins/gtkui/trkproperties.c]: This looks like a serious bug fix.. how to trigger the bug? I'm trying to figure out what's changed.. I will need to apply this fix to cocoaui too, I think (if it's valid)

I'd like to reproduce this, and be able to verify the fix.

**Comment by Lithopsian** [comment]
The bug is that once you have, or create, a field with more than 5,000 characters, you can't get rid of it.  Minor editing is prevented because the result might be a corruption, but removing the contents entirely should be possible so you can start again.

There was a commit a long time ago that created the line you highlight.  It was intended to allow removal of these fields.  I don't know if it worked then, but it doesn't work now so I make this change.

**Comment by Lithopsian** [comment]
@Alexey-Yakovenko is this whole workaround still needed?  If it is, is a tf field with more than 5,000 characters going to be a problem?  For example, a custom listview column for a long tag.

**Comment by Alexey-Yakovenko** [comment]
It's a workaround for GtkTreeView, and it's still needed. It crashes on some gtk versions when text is longer than certain amount of bytes or chars, far more than 5K though.

DdbListView is unaffected, therefore the workaround is not necessary there.

**Pull request merged**